plugins/connection/network_cli.py

```diff
@@ -1145,3 +1172,72 @@ def get_file(self, source=None, destination=None, proto="scp", timeout=30):
         elif proto == "sftp":
             with ssh.open_sftp() as sftp:
                 sftp.get(source, destination)
+
+    def send_command(
+        self,
+        command=None,
+        prompt=None,
+        answer=None,
+        sendonly=False,
+        newline=True,
+        prompt_retry_check=False,
+        check_all=False,
+        use_cache=False,
+    ):
+        """
+        This is a copy of CliconfBase send_command()
+        added to reduce dependency on ansible-base.
+
+        Executes a command over the device connection
+
+        This method will execute a command over the device connection and
+        return the results to the caller.  This method will also perform
+        logging of any commands based on the `nolog` argument.
+
+        :param command: The command to send over the connection to the device
+        :param prompt: A single regex pattern or a sequence of patterns to evaluate the expected prompt from the command
+        :param answer: The answer to respond with if the prompt is matched.
+        :param sendonly: Bool value that will send the command but not wait for a result.
+        :param newline: Bool value that will append the newline character to the command
+        :param prompt_retry_check: Bool value for trying to detect more prompts
+        :param check_all: Bool value to indicate if all the values in prompt sequence should be matched or any one of
+                          given prompt.
+        :param use_cache: Determines if command output should be cached and/or fetched from cache when single_user_mode
+                          is enabled.
+        :returns: The output from the device after executing the command
+        """
+        kwargs = {
+            "command": to_bytes(command),
```

Comment:
How about we add logic to invalid cache within the connection plugin.
1) By adding plugin configuration option to force invalid option
2) Introspect the `command` value to check if it starts with the platform-specific configure command (it can be a cliconf configurable option) which defaults to `configure`